src/vg/civcraft/mc/civmenu/TOSListener.java

```diff
@@ -66,13 +65,27 @@ public void run() {
 	public void playerMoveEvent(PlayerMoveEvent event) {
 		Player p = event.getPlayer();
 		if (!TOSManager.isTermPlayer(p, "CivMenu Agreement")) {
-			if(event.getTo().distance(locations.get(p.getUniqueId())) > plugin.GetConfig().get("terms.MovementRange").getInt()){
+			Location from = locations.get(p.getUniqueId());
+			if (from == null){
+				from = event.getFrom();
+				locations.put(p.getUniqueId(), from);
```

Comment:
This does solve the NPE, but not the issue behind it of players managing to slip through the login listener. Do we have any idea what could have possibly caused that?